frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts

```diff
@@ -302,31 +302,57 @@ const buildEChartsBarSeries = (
   };
 };
 
-function getShowSymbol(
-  seriesModel: SeriesModel,
-  seriesSettings: SeriesSettings,
+function getShowAutoSymbols(
+  seriesModels: SeriesModel[],
   dataset: ChartDataset,
+  seriesSettingsByDataKey: Record<string, SeriesSettings>,
   chartWidth: number,
-) {
-  // "line.marker_enabled" corresponds to the "Show dots on lines" series setting
-  // and can be true, false, or undefined
-  // true = on
-  // false = off
-  // undefined = auto
-  const isAuto = seriesSettings["line.marker_enabled"] == null;
-  if (!isAuto) {
-    return seriesSettings["line.marker_enabled"];
+): boolean {
+  if (chartWidth <= 0) {
+    return false;
   }
+
+  const seriesWithSymbols = seriesModels.filter(seriesModel => {
+    const seriesSettings = seriesSettingsByDataKey[seriesModel.dataKey];
+    if (["area", "line"].includes(seriesSettings.display ?? "")) {
```

Comment:
We can replace this entire block with just `return ["area", "line"].includes(seriesSettings.display ?? "")`